fiovb/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+project (fiovb C)
+
+set (SRC host/main.c)
+set (PRINTENV fiovb_printenv)
+set (SETENV fiovb_setenv)
+
+add_executable (${PROJECT_NAME} ${SRC})
+
+target_include_directories(${PROJECT_NAME}
+			   PRIVATE ta/include
+			   PRIVATE include)
+
+target_link_libraries (${PROJECT_NAME} PRIVATE teec)
+
+install (TARGETS ${PROJECT_NAME} DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+install (CODE "execute_process( \
+	       COMMAND ${CMAKE_COMMAND} -E create_symlink \
+	       ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/${PROJECT_NAME} \
+	       ${PRINTENV} \
+	       )"
+)
+
+install (CODE "execute_process( \
+	       COMMAND ${CMAKE_COMMAND} -E create_symlink \
+	       ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/${PROJECT_NAME} \
+	       ${SETENV} \
+	       )"
+)
+
+install (FILES ${CMAKE_CURRENT_BINARY_DIR}/../${PRINTENV}
```

Comment:
`${CMAKE_CURRENT_SOURCE_DIR}/${PRINTENV}` and ``${CMAKE_CURRENT_SOURCE_DIR}/${SETENV}`` is better because it will also work when the build directory is outside the source code.

Reply:
This is how it works now (CMAKE_BINARY_DIR != CMAKE_SOURCE_DIR in my setup).
CMAKE_CURRENT_BINARY_DIR is where both symlinks are created as a result of previous two install cmds (as current dir there is CMAKE_CURRENT_BINARY_DIR, not CMAKE_CURRENT_SOURCE_DIR)

Reply:
Maybe I didn't quite understand the problem, what I am trying to say is to check if it will support a structure like the following (used in bitbake where the build folder in not inside the source) :
```
.
├── build
└── optee-fiovb
    ├── CMakeLists.txt
    ├── fiovb
    │   ├── CMakeLists.txt
    │   ├── host
    │   │   ├── main.c
    │   │   ├── Makefile
    │   │   └── ta_fiovb.h -> ../ta/include/ta_fiovb.h
    │   ├── Makefile
    │   └── ta
    │       ├── entry.c
    │       ├── include
    │       │   └── ta_fiovb.h
    │       ├── Makefile
    │       ├── sub.mk
    │       ├── user_ta_header_defines.h
    │       └── user_ta.mk
    ├── LICENSE
    ├── Makefile
    └── README.md
```

Reply:
I tried locally but i can't because need more stuff
```
cd build
cmake ../optee-fiovb
-- The C compiler identification is GNU 12.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done (0.1s)
-- Generating done (0.0s)
-- Build files have been written to: /home/jq/foundries/git/optee-fiovb/top/build

make
[ 50%] Building C object fiovb/CMakeFiles/fiovb.dir/host/main.c.o
top/optee-fiovb/fiovb/host/main.c:11:10: fatal error: tee_client_api.h: No such file or directory
   11 | #include <tee_client_api.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

Reply:
nice, thanks for explaining.